### PR TITLE
accessControl & SOAP: Link website alert pages <-> help

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
Link website alert pages <-> help
SOAP: Update help links (Issue 8262)

## Related Issues
zaproxy/zaproxy#8189
zaproxy/zaproxy#8262

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [NA] Write tests
- [NA] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

**Comment by kingthorin** [comment]
I still need to do the SOAP part, but I wanted to open this to get some feedback on the Access Control bit because it's kind of a different beast.

File: addOns/accessControl/src/main/javahelp/org/zaproxy/zap/extension/accessControl/resources/help/contents/concepts.html [hunk 1]
```diff
@@ -68,6 +68,33 @@ <h3>Testing procedure</h3>
...
+<H3 id="id-10101">Access Control Issue - Improper Authentication</H3>
+
+    Insufficient Authentication occurs when a web site permits an attacker to access sensitive content or functionality without 
+    having to properly authenticate. Web-based administration tools are a good example of web sites providing access to sensitive functionality. 
+    Depending on the specific online resource, these web applications should not be directly accessible without requiring the user to properly verify 
+    their identity.
+    <br>
+    Alert ID: <a href="https://www.zaproxy.org/docs/alerts/10101/">10101</a>.
+
+<H3 id="id-10102>Access Control Issue - Improper Authorization</H3>
```
Comment (kingthorin) [inline addOns/accessControl/src/main/javahelp/org/zaproxy/zap/extension/accessControl/resources/help/contents/concepts.html:90]: Missing quote on id attr

File: addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.soap;
+
+public interface CommonScanRuleInfo {
+
+    default String getHelpLink() {
+        return "https://www.zaproxy.org/docs/desktop/addons/soap-support/#scanning";
```
Comment (thc202) [inline addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java:25]: Why not link to exact scan rule?

Comment (kingthorin) [reply addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java:25]: To the row in the table? Or are you picturing it displayed differently?

Comment (thc202) [reply addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java:25]: Row would be fine but maybe we should use the same format everywhere (I know GraphQL is using the same).

File: addOns/accessControl/src/main/javahelp/org/zaproxy/zap/extension/accessControl/resources/help/contents/concepts.html [hunk 3]
```diff
@@ -69,6 +69,32 @@ <h3>Testing procedure</h3>
 </ul>
 <strong>Note:</strong> Access control testing is not allowed in <code>Safe</code> mode nor <code>Protected</code> if the context is not in scope.
 
+<H2>Latest Code</H2> 
+
+<a href="https://github.com/zaproxy/zap-extensions/tree/main/addOns/accessControl">accessControl</a>
```
Comment (thc202) [inline addOns/accessControl/src/main/javahelp/org/zaproxy/zap/extension/accessControl/resources/help/contents/concepts.html:74]: Don't mind much the general one, for other scan rules we link directly to the scan rules' class.

Comment (thc202) [reply addOns/accessControl/src/main/javahelp/org/zaproxy/zap/extension/accessControl/resources/help/contents/concepts.html:74]: There is `AccessControlScannerThread` (that it does not extend the base scan rules does not make the code less relevant to the raised alerts).

File: addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java (hunk 2 above)
Comment (kingthorin) [reply addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java:25]: Okay, I made a separate Alerts page like GraphQL, and linked the rows.

File: addOns/soap/src/main/javahelp/org/zaproxy/zap/extension/soap/resources/help/contents/soap.html [hunk 4]
```diff
@@ -29,18 +29,27 @@ <h3>Form Handler Add-on Support</h3>
 <br>
 <img src="images/formHandlerExample.png" alt="Form Handler Example" width="300">
 
-<h2>Scanning</H2>
-The scan rules added by this add-on are:
-<ul>
-<li>Action Spoofing : <a href="http://www.ws-attacks.org/index.php/SOAPAction_Spoofing">http://www.ws-attacks.org/index.php/SOAPAction_Spoofing</a></li>
-<li>XML Injection : <a href="http://www.ws-attacks.org/index.php/XML_Injection">http://www.ws-attacks.org/index.php/XML_Injection</a></li>
```
Comment (kingthorin) [inline addOns/soap/src/main/javahelp/org/zaproxy/zap/extension/soap/resources/help/contents/soap.html:36]: This content is no longer available. I haven't yet checked the wayback machine.

**Comment by kingthorin** [comment]
Ready for another review.

File: addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.soap;
+
+public interface CommonScanRuleInfo {
```
Comment (thc202) [inline addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java:22]: I'd add the `getId()` and let the passive scan rule implement it.

File: addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    default String getHelpLink(int id) {
+        return "https://www.zaproxy.org/docs/desktop/addons/soap-support/alerts#id-" + id;
```
Comment (thc202) [inline addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java:25]: Worth adding `/` before the fragment to avoid a redirect.

**Comment by kingthorin** [comment]
Done, hopefully I did what you were picturing :grinning:

File: addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.soap;
+
+public interface CommonScanRuleInfo {
+
+    public int getId();
+
+    default String getHelpLink(int id) {
```
Comment (thc202) [inline addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/CommonScanRuleInfo.java:26]: Remove the parameter.

File: addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/SOAPXMLInjectionActiveScanRule.java [hunk 8]
```diff
@@ -207,4 +208,9 @@ public int getWascId() {
     public Map<String, String> getAlertTags() {
         return ALERT_TAGS;
     }
+
+    @Override
+    public String getHelpLink(int id) {
```
Comment (thc202) [inline addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/SOAPXMLInjectionActiveScanRule.java:213]: This can be removed too.

**Comment by kingthorin** [comment]
Done & done.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**